dtyper.py

```diff
@@ -1,23 +1,39 @@
-from dataclasses import make_dataclass, field
-from functools import wraps
+from __future__ import annotations
+
 import inspect
+from dataclasses import field, make_dataclass
+from functools import wraps
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Callable, Optional, Type, TypeVar, Union
+
+    from typing_extensions import ParamSpec
+
+    P = ParamSpec("P")
+    R = TypeVar("R")
 
 __version__ = '0.9.0'
 
 
 @wraps(make_dataclass)
-def dataclass(typer_command, base=None, **kwargs):
-    def to_field(k, v, default):
-        return (k, v) if default is ... else (k, v, field(default=default))
+def dataclass(
+    typer_command: Callable[P, R], base: Optional[Type] = None, **kwargs
```

Comment:
If there's this additional complexity, you should give each parameter its own line.